File: tests/CCSharp.Tests/Vector3Test.cs
using System;
using CCSharp.Attributes;
using CCSharp.ComputerCraft;

namespace CCSharp.Tests;

[LuaProgram]
public static class Vector3Test
{
    public static Vector3 testVector { get; set; } = new(5, 3, 2);
    
    [LuaMain]
    public static void Start()
    {
        Console.WriteLine("(" + testVector + ")");
        Console.WriteLine(testVector + new Vector3(4, 3, 1));
        Console.WriteLine(testVector * 4);
        Console.WriteLine(testVector / 2);
        Console.WriteLine(-testVector);
        Console.WriteLine(testVector == new Vector3(5, 3, 2));
        Console.WriteLine(testVector == new Vector3(4, 3, 1));
        Console.WriteLine(new Vector3());
    }
}

File: tests/CCSharp.Tests/lua/Vector3Test.lua
local testVector = vector.new(5, 3, 2);
function Start()
 local testVector = testVector;
 print(("("..tostring((testVector~=nil) and testVector or nil))..")")
 print(testVector+vector.new(4, 3, 1))
 print(testVector*4)
 print(testVector/2)
 print(-testVector)
 print(testVector==vector.new(5, 3, 2))
 print(testVector==vector.new(4, 3, 1))
 print(vector.new())
end
Start()